proxy/src/identity.rs

```diff
@@ -88,16 +88,12 @@ pub fn get(api: &coco::Api, id: &coco::Urn) -> Result<Identity, error::Error> {
 /// Retrieve the list of tracked identities.
 ///
 /// # Errors
-pub fn list_tracked_identities(api: &coco::Api) -> Result<Vec<Identity>, error::Error> {
-    let mut tracked_users = vec![];
-    for project in api.list_projects()? {
-        let project_urn = project.urn();
-        for peer in api.tracked(&project_urn)? {
-            tracked_users.push(peer.into())
-        }
-    }
-    // TODO(merle): Deduplicate `tracked_users`
-    Ok(tracked_users)
+pub fn list_tracked(api: &coco::Api) -> Result<Vec<Identity>, error::Error> {
+    let peers = api.list_projects()?.into_iter().flat_map(|project| {
+        api.tracked(&project.urn())
+            .expect("Peers could not be retrieved.")
```

Comment:
Ach! I always get annoyed when I remember iterators don't play well with `Result` :weary: I'd rather we avoid `expect` but not sure off the top of my head how you could do that nicely here.

Reply:
yeah, I couldn't figure out a different way :/

Reply:
Alternativelly drop the iterators and go for a classic loop, which allows to tap into the normal error infrastructure. The only other way is for the closure to return a result, which then needs to be flattened again and so on.

Reply:
@MeBrei had a classic loop and I suggested the iterator change :'( My bad...

Reply:
Stop tempting the youth with your FP promises. Poor souls!

Reply:
Can't wait for the next time I realise this point again and get annoyed.